components/kyma-environment-broker/internal/process/provisioning/ems_provision.go

```diff
@@ -0,0 +1,138 @@
+package provisioning
+
+import (
+	"fmt"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/servicemanager"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	EmsOfferingName = "enterprise-messaging"
+	EmsPlanName     = "default"
+)
+
+type EmsProvisionStep struct {
+	operationManager *process.ProvisionOperationManager
+}
+
+func NewEmsProvisionStep(os storage.Operations) *EmsProvisionStep {
+	return &EmsProvisionStep{
+		operationManager: process.NewProvisionOperationManager(os),
+	}
+}
+
+var _ Step = (*EmsProvisionStep)(nil)
+
+func (s *EmsProvisionStep) Name() string {
+	return "EMS_Provision"
+}
+
+func (s *EmsProvisionStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if operation.Ems.Instance.InstanceID != "" {
+		return operation, 0, nil
+	}
+
+	smCli, err := operation.ServiceManagerClient(log)
+	if err != nil {
+		return s.handleError(operation, err, log, "unable to create Service Manage client")
+	}
+	// get offering
+	offs, err := smCli.ListOfferingsByName(EmsOfferingName)
```

Comment:
there is a step for taking the offering IDs - ServiceManagerOfferingStep

Reply:
Refactored